src/main/java/org/apache/flink/benchmark/functions/LongSourceType.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.benchmark.functions;
+
+import org.apache.flink.api.common.eventtime.WatermarkStrategy;
+import org.apache.flink.api.connector.source.Boundedness;
+import org.apache.flink.streaming.api.datastream.DataStreamSource;
+import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
+
+import java.util.function.BiFunction;
+
+/**
+ * Enum based factory for different Long sources.
+ */
+public enum LongSourceType {
+	LEGACY_SOURCE((env, maxValue) -> {
+		return env.addSource(new LongSource(maxValue));
+	}),
+	FLIP_27_SOURCE_BOUNDED((env, maxValue) -> {
+		return env.fromSource(
+				new LongNewSource(Boundedness.BOUNDED, maxValue),
+				WatermarkStrategy.noWatermarks(),
+				"NewLongSource");
+	}),
+	FLIP_27_SOURCE_UNBOUNDED((env, maxValue) -> {
+		return env.fromSource(
+				new LongNewSource(Boundedness.BOUNDED, maxValue),
```

Comment:
`FLIP_27_SOURCE_UNBOUNDED` & `BOUNDED`? How does it differ compared to `FLIP_27_SOURCE_BOUNDED`? Do we need both? Is there a runtime difference between bounded & unbounded?

Reply:
Yes, for `BOUNDED` we might use the sorting `DataOutputs/Inputs`, use the pipelined scheduling, use different operators etc.

Reply:
By first question I meant that there is a bug with both BOUNDED and UNBOUNDED enum setting `Boundedness.BOUNDED` parameter.

Don't we have other benchmarks that are covering the case of sorting outputs/inputs?

Reply:
We do. I'd still have the enum here. I am fine though with dropping the `F27_BOUNDED` from `InputBenchmark`.

I spotted the bug after your comment and fixed it.